includes/Checker/Default_Check_Repository.php

```diff
@@ -99,37 +96,19 @@ public function get_checks( $flags = self::TYPE_ALL, array $check_slugs = array(
 			$checks += $this->runtime_checks;
 		}
 
-		// Filter out the specific check slugs requested.
-		if ( ! empty( $check_slugs ) ) {
-			$checks = array_map(
-				function ( $slug ) use ( $checks ) {
-					if ( ! isset( $checks[ $slug ] ) ) {
-						throw new Exception(
-							sprintf(
-								/* translators: %s: The Check slug. */
-								__( 'Check with the slug "%s" does not exist.', 'plugin-check' ),
-								$slug
-							)
-						);
-					}
-
-					return $checks[ $slug ];
-				},
-				$check_slugs
-			);
-		}
-
 		// Return all checks, including experimental if requested.
 		if ( $flags & self::INCLUDE_EXPERIMENTAL ) {
-			return $checks;
+			return new Default_Check_Collection( $checks );
 		}
 
 		// Remove experimental checks before returning.
-		return array_filter(
-			$checks,
-			static function ( $check ) {
-				return $check->get_stability() !== Check::STABILITY_EXPERIMENTAL;
-			}
+		return new Default_Check_Collection(
+			array_filter(
+				$checks,
+				static function ( $check ) {
+					return $check->get_stability() !== Check::STABILITY_EXPERIMENTAL;
+				}
+			)
```

Comment:
Would it make sense to add a `filter()` method on the `Check_Collection`?

Then we could do `return (new Default_Check_Collection( $checks ) )->filter( ... )`

Reply:
Added in https://github.com/WordPress/plugin-check/pull/315/commits/6b9df7fbe80fca772eb13737a29e9a43a750e206